src/guide/payment_notification.md

```diff
@@ -0,0 +1,115 @@
+# Receiving Payment Notifications via Webhook in Breez SDK
+
+## Overview
+The Breez SDK provides a robust feature for applications to receive real-time notifications for incoming payments over the Lightning Network. This is achieved through the use of webhooks. A webhook is a powerful tool that allows your application to be notified via a specified URL when a payment is received.
+
+## Setting Up the Webhook
+To utilize this feature, you need to set up a webhook that Breez can call when a payment is received.
+
+## Step 1: Define Your Webhook URL
+First, you need to define the URL endpoint in your application that will handle incoming payment notifications. This URL should be capable of receiving POST requests.
+
+## Step 2: Register the Webhook with Breez
+Next, register this URL with the Breez SDK. This step involves calling a specific SDK function.
+
+<custom-tabs category="lang">
+<div slot="title">Rust</div>
+<section>
+
+```rust,ignore
+{{#include ../../snippets/rust/src/webhook.rs:register-webook}}
+```
+</section>
+</custom-tabs>
+
+## Handling Incoming Notifications
+When a payment is received, Breez will send a POST request to your webhook URL. The request body will contain details about the payment.
+
+### Expected Webhook Payload
+The payment received webhook payload is json formatted and contains the following structure:
+
+<section>
+<pre>
+{
+ "type": "payment_received",
+ "data": {  
+  "payment_hash" : [payment hash]
+ }
+}
+</pre>
+</section>
+
+## Processing the Payload
+Processing the Payload is according to the application needs. One of the major use cases that can be handled via webhooks is receiving a payment in mobile app while the app is not running, or in short "offline receive"
+
+## Webhook integration for "offline recieve"
+In the context of mobile applications using the Breez SDK for Lightning Network payments, a crucial architectural component involves the seamless flow of payment notifications to the user's mobile device. This process is facilitated through a Notification Delivery Service (NDS), acting as an intermediary. When a payment is made to a user, the LSP sends a notification to the NDS, configured with a specific webhook URL. The NDS then processes this information and dispatches a push notification to the intended mobile device, ensuring the user receives timely updates about incoming payments. This architecture necessitates vendors to set up and maintain their own NDS, tailored to handle and forward these notifications efficiently.
+
+## Step 1: Run your own NDS
```

Comment:
Step 1 is also defined above, but they are not seperated by a header. I think these steps should start with 3 hashtags `###` and there should be a subheader seperating the sections.

Reply:
Done